networking/networking.go

```diff
@@ -50,6 +54,114 @@ type Networking struct {
 	nets   []activeNet
 }
 
+type NetConf struct {
+	plugin.NetConf
+	IPMasq bool `json:"ipMasq"`
+	MTU    int  `json:"mtu"`
+}
+
+func setupTapDevice() (string, *netlink.Link, error) {
+	ifName, err := tuntap.CreatePersistentIface(tuntap.Tap)
+	if err != nil {
+		return "", nil, fmt.Errorf("tuntap persist %v", err)
+	}
+	var link netlink.Link
+	link, err = netlink.LinkByName(ifName)
+	if err != nil {
+		return "", nil, fmt.Errorf("cannot find link %q: %v", ifName, err)
+	}
+	err = netlink.LinkSetUp(link)
+	if err != nil {
+		return "", nil, fmt.Errorf("cannot set link up %q: %v", ifName, err)
+	}
+	return ifName, &link, nil
+}
+
+func kvmSetupNetAddressing(network *Networking, n activeNet, ifName string) error {
+	// TODO: very ugly hack, that go through upper plugin, down to ipam plugin
+	n.conf.Type = n.conf.IPAM.Type
+	output, err := network.execNetPlugin("ADD", &n, ifName)
+	if err != nil {
+		return fmt.Errorf("problem executing network plugin %q (%q): %v", n.conf.Type, ifName, err)
+	}
+
+	result := plugin.Result{}
+	if err = json.Unmarshal(output, &result); err != nil {
+		return fmt.Errorf("error parsing %q result: %v", n.conf.Name, err)
+	}
+
+	if result.IP4 == nil {
+		return fmt.Errorf("net-plugin returned no IPv4 configuration")
+	}
+
+	n.runtime.IP, n.runtime.Mask, n.runtime.HostIP = result.IP4.IP.IP, net.IP(result.IP4.IP.Mask), result.IP4.Gateway
+	return nil
+}
+
+func KvmSetup(podRoot string, podID types.UUID, fps []ForwardedPort, privateNetList common.PrivateNetList, localConfig string) (*Networking, error) {
+	network := Networking{
+		podEnv: podEnv{
+			podRoot:      podRoot,
+			podID:        podID,
+			netsLoadList: privateNetList,
+			localConfig:  localConfig,
+		},
+	}
+	var e error
+	network.nets, e = network.loadNets()
+	if e != nil {
+		return nil, fmt.Errorf("error loading network definitions: %v", e)
+	}
+
+	for _, n := range network.nets {
+		switch n.conf.Type {
+		case "ptp":
+			ifName, link, err := setupTapDevice()
+			if err != nil {
+				return nil, err
+			}
+			n.runtime.IfName = ifName
+
+			err = kvmSetupNetAddressing(&network, n, ifName)
+			if err != nil {
+				return nil, err
+			}
+
+			// add address to host tap device
+			err = netlink.AddrAdd(
+				*link,
+				&netlink.Addr{
```

Comment:
This could be fixed by simply adding those labels, like:
```go
&netlink.Addr{
        *net.IPNet: &net.IPNet{},
        Label: ifName,
}
```
(Code is not tested, but I had similar issue when I refactoring image finding and fetching, https://github.com/coreos/rkt/blob/master/rkt/images.go#L58 and https://github.com/coreos/rkt/blob/master/rkt/images.go#L95)